tools/harness-automation/autothreadharness/harness_case.py

```diff
@@ -519,20 +519,23 @@ def _test_bed(self):
             logger.info('Topology config devices for case %s: %s', case_id, topo_mixed_devices)
             f_topo.close()
             needed_golden_devices = []
+            missing_golden_devices = topo_mixed_devices
             # mapping topology config devices with devices in settings
             for temp_mixed_device in topo_mixed_devices:
                 for temp_device in devices:
                     if temp_mixed_device[1] == temp_device[1]:
                         needed_golden_devices.append(temp_device)
                         devices.remove(temp_device)
+                        missing_golden_devices.remove(temp_mixed_device)
                         break
-            logger.info('Needed golden devices can be found in settings : %s', needed_golden_devices)
+            logger.info('Case-needed golden devices can be found in settings : %s', needed_golden_devices)
             if len(topo_mixed_devices) != len(needed_golden_devices):
+                logger.info('Missing golden devices : %s', missing_golden_devices)
```

Comment:
Statistics might make things easier. e.g.  Prompts "Missing 5 ARM devices and 1 OT device"

Reply:
will be logged as below if golden devices number is lower than topology config's:
INFO:autothreadharness.harness_case:Golden devices in topology config file for case 5.2.6: [('Leader', 'OpenThread'), ('Router_2', 'OpenThread'), ('Router_3', 'ARM'), ('Router_4', 'SiLabs'), ('Router_5', 'SiLabs'), ('Router_6', 'NXP'), ('Router_7', 'ARM'), ('Router_8', 'ARM'), ('Router_9', 'NXP'), ('Router_10', 'NXP'), ('Router_11', 'SiLabs'), ('Router_12', 'SiLabs'), ('Router_13', 'ARM'), ('Router_14', 'NXP'), ('Router_15', 'ARM'), ('Router_16', 'NXP'), ('Router_17', 'OpenThread'), ('Router_18', 'NXP'), ('Router_19', 'NXP'), ('Router_20', 'ARM'), ('Router_21', 'NXP'), ('Router_22', 'SiLabs'), ('Router_23', 'NXP')]
INFO:autothreadharness.harness_case:Golden devices in topology config file mapped in settings : [('COM21', 'OpenThread'), ('COM13', 'ARM'), ('COM5', 'SiLabs'), ('COM14', 'ARM'), ('COM31', 'NXP'), ('COM16', 'SiLabs'), ('COM26', 'ARM'), ('COM24', 'NXP'), ('COM28', 'NXP'), ('COM10', 'NXP'), ('COM12', 'NXP')]
INFO:autothreadharness.harness_case:Missing Devices: {'OpenThread': 2, 'ARM': 3, 'SiLabs': 3, 'NXP': 4}